### PR TITLE
OOP RUBY - User interface

### PR DESCRIPTION
**Pull request by teekaytech**
### In this milestone, I did the following

- Created a new branch called user_interface
- Added contents to main.rb file for inputs and outputs
- Mimic the basic flow of tic tac toe game
- Have a deeper understanding of the game flow

## Kindly review for approval.

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,2 +1,54 @@
...
+# A game board will be displayed to the user, consisting of numbers from 1 to 9
+
+# We also have arrays that stores the values of selected moves by the players
+p1_move = []
+p2_move = []
+
+puts "\nWith each boxes displayed from 1 to 9, each player can type in a value >=1 and <= 9"
+
+count = 0
+while count < 9
```
Comment (shshamim63) [inline bin/main.rb]: [Suggestion for your next milestone]
Your loop should not depend on any constant number. Cause it reduces the scalability of your code.

File: bin/main.rb [hunk 2]
```diff
@@ -1,2 +1,54 @@
...
+      print "#{player_two}, select your move: "
+      input = gets.chomp.to_i
+      if (input >= 1) && (input <= 9)
+        p2_move << input
+        break
+      else
+        print 'Invalid input, try again! '
+      end
+    end
+  end
```
Comment (shshamim63) [inline bin/main.rb]: [Optional]
both if and else contains the same code except the variables. It is the right place to introduce a method to cover this task.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :1234: Required Changes :1234: 
___
Hi @teekaytech :wave: 
Almost there :clap:  only one point is blocking me to approve this milestone. Let's complete it,
___
### :spades: Mandatory Changes :spades: 
___
1. Well, the game should inform if there is a winner or it is a draw, but I can not see anything mentioned related to that. So please cover this scenario.
___
#### Please fix the mentioned issue and send the final review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

File: bin/main.rb (hunk 2 above)
Comment (teekaytech) [reply bin/main.rb]: I already have a plan to change this when I start the game logic in milestone 3.

File: bin/main.rb (hunk 1 above)
Comment (teekaytech) [reply bin/main.rb]: This will be taken care of in the next milestone.

**Comment by teekaytech** [comment]
### Thank you, @shshamim63 for our response.

I have added the required scenario from your previous review.

Kindly review again, for approval.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @teekaytech      
#### Great Job so far, but there are a few more things required. 
* [x] `[REQUIRED]` game informs player if selected move is invalid
* [x] `[REQUIRED]` game displays board after player move
* [x] `[REQUIRED]` game informs player if selected move is a winning move
* [x] `[REQUIRED]` game informs player if selected move is a draw move
- [x]  Check comments below

### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: bin/main.rb [hunk 3]
```diff
@@ -1,2 +1,60 @@
...
+# A game board will be displayed to the user, consisting of numbers from 1 to 9
+
+# We also have arrays that stores the values of selected moves by the players
+p1_move = []
+p2_move = []
+
+puts "\nWith each boxes displayed from 1 to 9, each player can type in a value >=1 and <= 9"
+
+count = 0
+while count < 9
```
Comment (kobiyoyo) [inline bin/main.rb]: As previous TSE specified you need to use a variable to terminate the loop not a number.for example
````ruby

game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
`````

Comment (teekaytech) [reply bin/main.rb]: Fixed.

**Comment by teekaytech** [comment]
### Thank you, @kobiyoyo  for your response.
I have fixed all the required issues raised. I did the following:
- I ensured that the game informs the players if the selected move is invalid
- I also ensured that the game displays board after the player move
- I added the feature for the game to inform the players if the selected move is a winning move
- In addition, I ensured that the game informs the player if it ended in a draw.
- Lastly, I used a variable to terminate the loop, not a number.

Kindly review again, for approval. :wavy_dash:

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Well done job :smile: You can merge it :+1: 
Good continuation :handshake: 

Mael FOSSO

**Pull request merged**